src/cata_tiles.cpp

```diff
@@ -1675,6 +1675,45 @@ const tile_type *cata_tiles::find_tile_looks_like( std::string &id, TILE_CATEGOR
     return nullptr;
 }
 
+bool cata_tiles::find_overlay_looks_like( const bool male, const std::string &overlay,
+        std::string &draw_id )
+{
+    bool exists = false;
+
+    std::string looks_like;
+    std::string over_type;
+    if( overlay.substr( 0, 5 ) == "worn_" ) {
+        looks_like = overlay.substr( 5 );
+        over_type = "worn_";
+    } else if( overlay.substr( 0, 8 ) == "wielded_" ) {
+        looks_like = overlay.substr( 8 );
+        over_type = "wielded_";
+    } else {
+        looks_like = overlay;
+    }
+
+    int cnt = 0;
+    while( !looks_like.empty() && cnt < 10 ) {
+        draw_id = ( male ? "overlay_male_" : "overlay_female" ) + over_type + looks_like;
+        if( tileset_ptr->find_tile_type( draw_id ) ) {
+            exists = true;
+            break;
+        }
+        draw_id = "overlay_" + over_type + looks_like;
+        if( tileset_ptr->find_tile_type( draw_id ) ) {
+            exists = true;
+            break;
+        }
+        if( !item::type_is_defined( looks_like ) ) {
+            break;
+        }
+        const itype *new_it = item::find_type( looks_like );
+        looks_like = new_it->looks_like;
+        cnt += 1;
```

Comment:
Should this just be a for loop?

Reply:
yeah, in this instance, that would make sense. thanks!